torch/onnx/_internal/exporter.py

```diff
@@ -882,29 +895,38 @@ def adapt_torch_inputs_to_onnx(
             >>> onnx_program = torch.onnx.dynamo_export(func_nested_input, x_dict, y_tuple)
             >>> print(x_dict, y_tuple)
             {'a': tensor(1.)} (tensor(2.), (tensor(3.), tensor(4.)))
-            >>> print(onnx_program.adapt_torch_inputs_to_onnx(x_dict, y_tuple, model_with_state_dict=func_nested_input))
+            >>> print(onnx_program.adapt_torch_inputs_to_onnx(x_dict, y_tuple))
             (tensor(1.), tensor(2.), tensor(3.), tensor(4.))
 
         .. warning::
             This API is experimental and is *NOT* backward-compatible.
 
         """
-        # model specified by the user has precedence, when specified
-        model_with_state_dict = model_with_state_dict or self._model_torch
-        assert (
-            model_with_state_dict is not None
-        ), "model_with_state_dict must be specified."
+        # backward compatibility: Respect the usage of *args, **kwargs for users and
+        # also None for the optional arguments
+        if (
+            "exportped_program" in model_kwargs
+            and model_kwargs["exportped_program"] is None
+        ):
+            exported_program = model_kwargs.pop("exportped_program")
+        if (
+            "onnx_input_names" in model_kwargs
+            and model_kwargs["onnx_input_names"] is None
+        ):
+            onnx_input_names = model_kwargs.pop("onnx_input_names")
+
         return self._input_adapter.apply(
-            *model_args, model=model_with_state_dict, **model_kwargs
+            *model_args,
+            model=exported_program,
```

Comment:
Is this the place you mentioned in https://github.com/pytorch/pytorch/pull/119211/files#r1480511928, where as the model argument is not passed in for other type of graph extractors?

Reply:
Oh it's inside  torch_export_graph_extractor that I use option.fake_context is not None to conditionally select the appended adapters.

Reply:
I didn't articulate the reason changing it. 

https://github.com/pytorch/pytorch/blob/91d1d2c421d0e94c8d3696dc27a44b466e34cf3e/torch/onnx/_internal/exporter.py#L711

I just wanted to make it clear that nn.Module doesn't need adapter because it is not prociding anything during adapting.